File: DAA/factorial.c
# include <stdio.h>
# include <time.h>

int factorial(int);

int main()
{
    clock_t begin = clock();

    int num, fac;
    printf("Enter the number you need factorial of: ");
    scanf("%d", &num);

    fac = factorial(num);
    printf("Factorial of %d = %d\n\n", num, fac);

    clock_t end = clock();

    double time = (double)(end - begin) / CLOCKS_PER_SEC;

    printf("Time taken to execute: %f", time);
    return 0;
}

int factorial (int n)
{
    int i, sum = 1;
    for (i = n; i > 0; i--)
    {
        sum = sum * i;
    }
    return sum;
}

File: Java/factorial.java
import java.util.*;

public class Factorial {
    public static void main(String[] args) {
        int fac, n;
        fac = 1;

        try(Scanner scan = new Scanner(System.in);)
        {
        System.out.println("Enter the number you need factorial for:" );
        n = scan.nextInt();
        scan.close();

        for(int i = 1; i <= n; i++)
        {
            fac *= i;
        }

        System.out.println(n + "! = "+ fac);
    }
}
}